src/main/kotlin/org/jetbrains/exposed/sql/Constraints.kt

```diff
@@ -61,6 +61,36 @@ data class ForeignKeyConstraint(val fkName: String, val refereeTable: String, va
 
 }
 
+data class CheckConstraint(val tableName: String, val checkName: String, val checkOp: String) : DdlAware {
+
+    companion object {
+        fun from(table: Table, name: String, op: Op<Boolean>): CheckConstraint {
+            val tr = TransactionManager.current()
+            return CheckConstraint(tr.identity(table), if (name.isBlank()) "" else tr.quoteIfNecessary(name), op.toString())
+        }
+    }
+
+    internal val checkPart = buildString {
+        if (checkName.isNotBlank()) append(" CONSTRAINT $checkName")
+        append(" CHECK ($checkOp)")
+    }
+
+    override fun createStatement(): List<String> {
+        if (currentDialect is MysqlDialect) throw UnsupportedOperationException("Check constraints are not currently supported by MySQL")
```

Comment:
I guess it's better to log warn or error, because MySQL parses `CHECK` expression, but ignore it. 
If you work with different databases and use same Exposed classes then you will have problem running it on MySQL.